backend/src/auth/dto/intra-tokendata.dto.ts

```diff
@@ -0,0 +1,26 @@
+import { Equals, IsNotEmpty, IsNumber, IsOptional } from 'class-validator';
+
+export class IntraTokendataDto {
+	@IsNotEmpty()
+	@Equals('bearer')
+	token_type: string;
+
+	@IsNotEmpty()
+	access_token: string;
+
+	@IsOptional()
+	@IsNotEmpty()
+	refresh_token?: string;
+
+	@IsOptional()
+	@IsNotEmpty()
+	scope?: string;
+
```

Comment:
curious what the scope does or mean in this one?

Reply:
from:  https://api.intra.42.fr/apidoc/guides/web_application_flow
> A space separated list of [scopes](https://api.intra.42.fr/apidoc/guides/web_application_flow#scopes). If not provided, scope defaults to an empty list of scopes for users that don't have a valid token for the app. For users who do already have a valid token for the app, the user won't be shown the OAuth authorization page with the list of scopes. Instead, this step of the flow will automatically complete with the same scopes that were used last time the user completed the flow.

We don't use it, but it exists in the response of the api..